typescript/schedules.md

```diff
@@ -0,0 +1,110 @@
+# Schedules
+
+- [Docs](https://docs.temporal.io/workflows/#schedules)
+- gRPC API:
+  - [methods](https://github.com/temporalio/api/blob/799926c86eb13d8a9717d3561ab9b0df43796c06/temporal/api/workflowservice/v1/service.proto#L328-L370)
+  - [`request_response.proto`](https://github.com/temporalio/api/blob/799926c86eb13d8a9717d3561ab9b0df43796c06/temporal/api/workflowservice/v1/request_response.proto#L821-L957)
+  - [`schedule/v1/message.proto`](https://github.com/temporalio/api/blob/master/temporal/api/schedule/v1/message.proto)
+  - [`ScheduleOverlapPolicy` enum](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/schedule.proto)
+
+## TS API
+
+```ts
+import { ScheduleClient, ScheduleOverlapPolicy } from '@temporalio/client'
+import { addWeeks } from 'date-fns'
+import { myWorkflow } from './workflows'
+
+const client = new ScheduleClient()
+
+const schedule = await client.create({
+  id: 'biz-id',
+  spec: {
+    // every hour at minute 5
+    interval: {
+      every: '1h',
+      at: '5m',
+    },
+    // every 20 days since epoch at day 2
+    // interval: {
+    //   every: '20d',
+    //   at: '2d'
+    // }
+    exclude: {
+      // skip 11:05 pm
+      hour: 23,
+      minute: 5,
+    },
+    endAt: addWeeks(new Date(), 4), 
+    jitter: '30s',
+    timezone: 'US/Eastern',
+  },
+  action: {
+    startWorkflow: {
+      workflowId: 'biz-id',
+      type: myWorkflow,
+      args: ['sorry this is the only thing reused, chad 😄'],
+    },
+  },
+  policies: {
+    overlap: ScheduleOverlapPolicy.BUFFER_ONE,
+    catchupWindow: '2m',
+    pauseOnFailure: true,
+  },
+  state: {
+    note: 'started schedule',
+    paused: true,
+    limitedActions: 10,
+  },
+  patch: {
+    triggerImmediately: true,
+    backfill: [
+      {
+        start: new Date(),
+        end: new Date(),
+        overlap: ScheduleOverlapPolicy.ALLOW_ALL,
+      },
+    ],
+    pause: true, // redundant with state.paused above (can use either)
+  },
+  memo,
+  searchAttributes,
+})
+
+const scheduleDescription = await schedule.describe()
+
+const matchingStartTimes = await schedule.listMatchingTimes({ start: new Date(), end: new Date() })
+
+await schedule.update({
+  spec,
+  action,
+  policies,
+  state,
+  conflictToken: scheduleDescription.conflictToken,
+})
+
+await schedule.patch({
+  triggerImmediately: true,
+  backfill,
+  unpause: true,
```

Comment:
Why is pause resume done in both patch and update(state)?  @dnr

Reply:
@lorensr I think adding top handle pause and resume methods would be very convenient and possibly backfill too. Not sure I would expose patch at all in the high level api

Reply:
> Not sure I would expose patch at all in the high level api

Is there no reason a user may want to do multiple of the patch things at once? What's the fear of exposing this?

Reply:
No "fear" just wanting to simplify, not sure how often a user would want to patch multiple things compared to backfill / toggle pause state.

Reply:
Replacing:

```ts
await schedule.patch({
  triggerImmediately: true,
  backfill,
  unpause: true,
})
```

with:

```ts
await schedule.trigger()
await schedule.backfill(backfill) // also takes array
await schedule.pause('note: pausing')
await schedule.unpause('now unpause')
```

I don't imagine wanting to patch multiple things at once. If needed, can call multiple individual methods, or can use raw gRPC service.